src/node.cc

```diff
@@ -1272,6 +1276,129 @@ void SetupNextTick(const FunctionCallbackInfo<Value>& args) {
+struct promise_abort_info {
+  int fd;
+  pid_t pid;
+  int refcount;
+};
+
+void OpenCoreDump(Environment* env, Local<Promise> promise) {
+  if (!abort_on_unhandled_rejection) return;
+#ifdef __POSIX__
+  struct rlimit lim;
+  CHECK_EQ(getrlimit(RLIMIT_CORE, &lim), 0);
+  if (lim.rlim_cur == 0) {
+    // Don't bother creating processes if we cannot create core dumps anyways.
+    return;
+  }
+  // Look up whether there is already a core dump holder process for this
+  // rejection reason, to better capture failure propagation along Promise
+  // chains.
+  Local<NativeWeakMap> reason_map = env->promise_reject_reason_map();
+  if (reason_map.IsEmpty()) {
+    reason_map = NativeWeakMap::New(env->isolate());
+    env->set_promise_reject_reason_map(reason_map);
+  }
+  CHECK_EQ(promise->State(), Promise::kRejected);
+  Local<Value> other = reason_map->Get(promise->Result());
+  if (other->IsPromise()) {
+    Local<Value> uint8array =
+        other.As<Promise>()->GetPrivate(env->context(),
+                                        env->promise_abort_info_symbol())
+                                            .ToLocalChecked();
+    CHECK(uint8array->IsUint8Array());
+    promise_abort_info* info =
+        reinterpret_cast<promise_abort_info*>(Buffer::Data(uint8array));
+    info->refcount++;
+    promise->SetPrivate(env->context(),
+                        env->promise_abort_info_symbol(),
+                        uint8array).FromJust();
+    return;
+  }
+  reason_map->Set(promise->Result(), promise);
+  int pipes[2] = { -1, -1 };
+  pid_t pid;
+
+  if (pipe(pipes) == -1) return;
+  pid = fork();
+  if (pid == -1) {
+    close(pipes[0]);
+    close(pipes[1]);
+    return;
+  }
+  if (pid == 0) {
+    bool do_abort;
+    int rc = read(pipes[0], &do_abort, sizeof(do_abort));
+    if (rc > 0 && !do_abort)
+      _exit(0);
+    else
+      abort();
+  }
+  close(pipes[0]);
+
+  Local<ArrayBuffer> ab =
+      ArrayBuffer::New(env->isolate(), sizeof(promise_abort_info));
+  Local<Value> uint8array =
+      Uint8Array::New(ab, 0, sizeof(promise_abort_info));
+  promise_abort_info* info =
+      reinterpret_cast<promise_abort_info*>(Buffer::Data(uint8array));
+  promise->SetPrivate(env->context(),
+                      env->promise_abort_info_symbol(),
+                      uint8array).FromJust();
+  info->fd = pipes[1];
+  info->pid = pid;
+  info->refcount = 1;
+#endif  // __POSIX__
+}
+
+void CloseCoreDump(Environment* env, Local<Promise> promise, bool do_abort) {
+  if (!abort_on_unhandled_rejection) return;
+#ifdef __POSIX__
+  CHECK_EQ(promise->State(), Promise::kRejected);
+  Local<Value> uint8array =
+      promise->GetPrivate(env->context(),
+                          env->promise_abort_info_symbol()).ToLocalChecked();
+  if (!uint8array->IsUint8Array()) {
+    if (do_abort) {
+      // This may happen when e.g. fork()ing itself failed due to resource
+      // constraints.
+      abort();
```

Comment:
`ABORT()`, here and elsewhere?  It prints a stack trace.

Reply:
I’ve added it here, but the other instances of `abort();` are used for an “expected” abort after printing the Promise’s stack trace.